### DISCUSSION TITLE
Unexpected scroll behavior when itemSize is automatically estimated

### DISCUSSION THREAD

**Discussion by mariia-mitiureva** [inokawa/virtua#500] [General]
Hi, I noticed some weird behavior, not sure if it's an issue or not.

I modified Chat default story [like this](https://gist.github.com/mariia-mitiureva/45fce0b19b700bb8466c64ae380312bc)

So, on the first render, there is one big item in the list, after a few seconds some smaller items are prepended. After that, if I scroll up just a bit, scroll will jump to the very top

https://github.com/user-attachments/assets/12fe8e66-9779-4322-8934-d77f218452d0

The issue can be fixed by setting `itemSize` to some small number, 30 for example. Is it expected or is there some issue with automatically estimated item size? I see, that it's recommended to rely on automatically estimated sizes..

**Comment by inokawa** [comment]
In auto estimation mode, default item size will be estimated when the sum of item sizes exceed the viewport size for the first time. Probably the size of the huge one item was treated as default size in that situation.

**Reply by mariia-mitiureva** [reply]
So is it expected behavior or is it possible to improve this?

Also, when I set `itemSize` to some small number, another issue gets worse. 
When list items are quite heavy components, and their size if greater than `itemSize` ([code snippet](https://gist.github.com/mariia-mitiureva/0054c377e86bbafd52f46b7857ef724e)), the initial scroll to bottom is quite noticeable.

https://github.com/user-attachments/assets/f5e4403b-a481-4234-9c4d-cb5bc6abce7b

In auto estimation mode it happens too, but less often.
